src/stateful-event-subscriber.ts

```diff
@@ -134,23 +246,44 @@ export abstract class StatefulEventSubscriber<State>
       index = indexBlockEnd;
     }
     this.invalid = false;
+
+    if (
+      !this.dexHelper.config.isSlave &&
+      this.masterPoolNeeded &&
+      this.state === null &&
+      blockNumberForMissingStateRegen
+    ) {
+      const createNewState = async () => {
+        this.logger.debug(
+          `${this.parentName}: ${this.name}: master generate new state because state is null`,
+        );
+        const state = await this.generateState(blockNumberForMissingStateRegen);
+        this.setState(state, blockNumberForMissingStateRegen);
+      };
+      createNewState();
```

Comment:
If we are not awaiting it, I think it worth to catch the error in case of unexpected situation and to not receive unhandled promise rejection